rpcserver.go

```diff
@@ -4766,63 +4766,61 @@ func (r *rpcServer) ExportChannelBackup(ctx context.Context,
-	// nothing to verify.
-	if in.GetSingleChanBackups() == nil && in.GetMultiChanBackup() == nil {
-		return nil, fmt.Errorf("either a Single or Multi channel " +
+	valid := false
+	switch {
+	// If neither a Single or Multi has been specified, then we have nothing
+	// to verify.
+	case in.GetSingleChanBackups() == nil && in.GetMultiChanBackup() == nil:
+		return nil, errors.New("either a Single or Multi channel " +
 			"backup must be specified")
-	}
 
-	// If a Single is specified then we'll only accept one of them to allow
-	// the caller to map the valid/invalid state for each individual
-	// Single.
-	if in.GetSingleChanBackups() != nil &&
-		len(in.GetSingleChanBackups().ChanBackups) != 1 {
+	// Either a Single or a Multi must be specified, but not both.
+	case in.GetSingleChanBackups() == nil && in.GetMultiChanBackup() == nil:
+		return nil, errors.New("either a Single or Multi channel " +
+			"backup must be specified, but not both")
 
-		return nil, fmt.Errorf("only one Single is accepted at a time")
-	}
-
-	// By default, we'll assume that both backups are valid.
-	resp := lnrpc.VerifyChanBackupResponse{
-		SinglesValid: true,
-		MultiValid:   true,
-	}
+	// If a Single is specified then we'll only accept one of them to allow
+	// the caller to map the valid/invalid state for each individual Single.
+	case in.GetSingleChanBackups() != nil:
+		chanBackupsProtos := in.GetSingleChanBackups().ChanBackups
+		if len(chanBackupsProtos) != 1 {
+			return nil, errors.New("only one Single is accepted " +
+				"at a time")
+		}
 
-	if in.GetSingleChanBackups() != nil {
-		// First, we'll convert the raw byte sliice into a type we can
+		// First, we'll convert the raw byte slice into a type we can
 		// work with a bit better.
-		chanBackupsProtos := in.GetSingleChanBackups().ChanBackups
 		chanBackup := chanbackup.PackedSingles(
 			[][]byte{chanBackupsProtos[0].ChanBackup},
 		)
 
 		// With our PackedSingles created, we'll attempt to unpack the
-		// backup. If this fails, then we know the backup is invalid
-		// for some reason.
+		// backup. If this fails, then we know the backup is invalid for
+		// some reason.
 		_, err := chanBackup.Unpack(r.server.cc.keyRing)
-		resp.SinglesValid = err == nil
-	}
+		valid = err == nil
 
-	if in.GetMultiChanBackup() != nil {
-		// Similarly, we'll convert the raw byte slice into a
-		// PackedMulti that we can easily work with.
+	case in.GetMultiChanBackup() != nil:
+		// We'll convert the raw byte slice into a PackedMulti that we
+		// can easily work with.
 		packedMultiBackup := in.GetMultiChanBackup().MultiChanBackup
 		packedMulti := chanbackup.PackedMulti(packedMultiBackup)
 
-		// We'll now attempt to unpack the Multi. If this fails, then
-		// we know it's invalid.
+		// We'll now attempt to unpack the Multi. If this fails, then we
+		// know it's invalid.
 		_, err := packedMulti.Unpack(r.server.cc.keyRing)
-		resp.MultiValid = err == nil
+		valid = err == nil
```

Comment:
might be better to just return the error directly, and the user can infer that it's valid if there was no error

Reply:
Fixed.